src/plugin.js

```diff
@@ -391,191 +455,155 @@ const initPlugin = function(player, options) {
+    this.player_.addChild('BigVrPlayButton', {}, this.bigPlayButtonIndex_);
+    this.player_.bigPlayButton = this.player_.getChild('BigVrPlayButton');
     // mobile devices
     if (videojs.browser.IS_ANDROID || videojs.browser.IS_IOS) {
-      player.controlBar.addChild('CardboardButton', {});
+      this.player_.controlBar.addChild('CardboardButton', {});
     }
 
     // if ios remove full screen toggle
     if (videojs.browser.IS_IOS) {
-      player.controlBar.fullscreenToggle.hide();
+      this.player_.controlBar.fullscreenToggle.hide();
     }
 
-    player.vr.camera.position.set(0, 0, 0);
-
-    player.vr.renderer = new THREE.WebGLRenderer({
+    this.camera.position.set(0, 0, 0);
+    this.renderer = new THREE.WebGLRenderer({
       devicePixelRatio: window.devicePixelRatio,
       alpha: false,
       clearColor: 0xffffff,
       antialias: true
     });
 
-    player.vr.renderer.setSize(player.currentWidth(), player.currentHeight());
-    player.vr.effect = new VREffect(player.vr.renderer);
+    this.renderer.setSize(this.player_.currentWidth(), this.player_.currentHeight());
+    this.effect = new VREffect(this.renderer);
 
-    player.vr.effect.setSize(player.currentWidth(), player.currentHeight());
-    player.vr.vrDisplay = null;
+    this.effect.setSize(this.player_.currentWidth(), this.player_.currentHeight());
+    this.vrDisplay = null;
 
     // Previous timestamps for gamepad updates
-    player.vr.prevTimestamps_ = [];
+    this.prevTimestamps_ = [];
 
-    player.vr.manager = new WebVRManager(player.vr.renderer, player.vr.effect, {hideButton: true});
+    this.manager = new WebVRManager(this.renderer, this.effect, {hideButton: true});
 
-    player.vr.renderedCanvas = player.vr.renderer.domElement;
+    this.renderedCanvas = this.renderer.domElement;
 
-    player.vr.renderedCanvas.style.width = 'inherit';
-    player.vr.renderedCanvas.style.height = 'inherit';
+    this.renderedCanvas.style.width = 'inherit';
+    this.renderedCanvas.style.height = 'inherit';
 
-    container.insertBefore(player.vr.renderedCanvas, container.firstChild);
-    videoEl.style.display = 'none';
+    this.player_.el().insertBefore(this.renderedCanvas, this.player_.el().firstChild);
+    this.getVideoEl_().style.display = 'none';
 
-    // Handle window resizes
-    function onWindowResize(event) {
-      const width = player.currentWidth();
-      const height = player.currentHeight();
-
-      player.vr.effect.setSize(width, height);
-      player.vr.camera.aspect = width / height;
-      player.vr.camera.updateProjectionMatrix();
+    if (window.navigator.getVRDisplays) {
+      window.navigator.getVRDisplays().then((displays) => {
+        if (displays.length > 0) {
+          this.log('WebVR supported, VRDisplays found.');
+          this.vrDisplay = displays[0];
+          this.log(this.vrDisplay);
+        } else {
+          this.triggerError({code: 'web-vr-no-devices-found', dismiss: false});
```

Comment:
There are a few cases where this is being called as `triggerError`, but it appears the function is `triggerError_`. It seems it would... ironically... trigger an error! 😆